### PR TITLE
Rewriting and tests

### PR DESCRIPTION
**Pull request by ulasonat**

### CODE REVIEW

File: R/Utilities.R [hunk 1]
```diff
@@ -124,27 +124,58 @@ function(pm,ki,byrow)
...
-	    if(length(j)>0) {
-        if((!all(j<0))&(!all(j>0))) stop("The column selection indices have different signs!")
-        for(k in 1:length(pd$dlist)) 
-          pd$dlist[[k]]<-pd$dlist[[k]][j]# retain or delet  the j.th column from the k.th row
-        pd$dim[2] <- if(j[1]<0) pd$dim[2]-length(j) else length(j)  } }
-    pd$degree<-pd$degree[if(length(i)) i else 1:pd$dim[1],if(length(j)) j else 1:pd$dim[2],drop=FALSE] 
-    pm<-polyMconvert(pd,class(pm)[1])		
-    return(pm) 
-   }
+	  if (length(i) > 0) {
```
Comment (namezys) [inline R/Utilities.R]: invalid indent. probably, you use tab instead of space

File: R/Utilities.R [hunk 2]
```diff
@@ -124,27 +124,58 @@ function(pm,ki,byrow)
...
+        pd$dim[1] <- length(i)
+      }
+    } 
+  }
+  
+  if (length(j) > 0) { 
+    j <- j[j!=0]
+    j <- j[j<=pd$dim[2]]
+    
+	  if (length(j) > 0) {
```
Comment (namezys) [inline R/Utilities.R]: tabs?

Comment (ulasonat) [reply R/Utilities.R]: shouldn't I use tab inside if?

Comment (namezys) [reply R/Utilities.R]: this line contains a mix of space and tabs. You have to use only spaces.

**Comment by ulasonat** [comment]
should I squash?

**Pull request merged**